### PR TITLE
update world_size and process_index for SMP 

### PR DESCRIPTION
**Pull request by cavdard**
This PR updates world_size and process_index for SMP when prescaled_batch is True/False.

### CODE REVIEW

**Comment by rahul003** [comment]
Relevant details https://github.com/huggingface/transformers/pull/15809#issuecomment-1050198829

**Comment by rahul003** [comment]
Prescaled batch is a new construct in SMP. SMP does tensor parallelism within a DP group. What this means is that it shuffles around the samples read by all ranks so that the rank which owns the shard of a layer/parameter can see relevant portion of samples for all samples across the data parallel group. To reduce the communication before layer execution, and to support cases where we can't use a large batch size (i.e. all ranks can't get their own sample), we recommend using prescaled batch, where all ranks in same rdp group read the same batch of data. Hence this usage affects the number of examples seen. From data loading perspective, the number of ranks which read different data changes in prescaled and non-prescaled case. World size and process index in transformers seem to relate to number of times data parallelism happens in the job, hence this change.

More details here
https://sagemaker.readthedocs.io/en/stable/api/training/smd_model_parallel_general.html#prescaled-batch

File: src/transformers/training_args.py [hunk 1]
```diff
@@ -1248,7 +1248,10 @@ def main_process_first(self, local=True, desc="work"):
                 is_main_process = self.local_process_index == 0
                 main_process_desc = "main local process"
             else:
-                is_main_process = self.process_index == 0
+                if is_sagemaker_mp_enabled():
+                    is_main_process = smp.rank() == 0
+                else:
+                    is_main_process = self.process_index == 0
```
Comment (sgugger) [inline src/transformers/training_args.py:1254]: This should be grouped with the previous else in an `elif` block

**Review by sgugger — APPROVED** [review]
Thanks for your PR, just left a nit on styling.

**Pull request closed without merging**